src/data/set/basic.lean

```diff
@@ -1,25 +1,73 @@
 /-
 Copyright (c) 2014 Jeremy Avigad. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Author: Jeremy Avigad, Leonardo de Moura
+Authors: Jeremy Avigad, Leonardo de Moura
 -/
+
 import tactic.basic tactic.finish data.subtype logic.unique
-open function
 
-/-! # Basic properties of sets
+/-!
+
+# Basic properties of sets
+
+A set in Lean is a subset of a type. This file provides some basic definitions related to sets
+and functions not present in the core library, as well as extra lemmas for functions in the core
+library (empty set, univ, union, intersection, insert, singleton, set-theoretic difference,
+complement, and powerset).
+
+## Main definitions
+
+* `strict_subset s t` : the predicate `s ⊆ t` but `s ≠ t`.
+
+* `nonempty s` : the predicate `s ≠ ∅`. Note that this is the canonical way to express the fact that
```

Comment:
I think that sentence is cryptic. I understand it only because I already know its content. Maybe "preferred" is better than canonical here, and you can refer to the implementation notes (or move what is there here).